server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -289,6 +294,39 @@ public boolean rename(Path path, Path newPath) throws IOException {
     return source.rename(path, newPath);
   }
 
+  @Override
+  public List<Future<Boolean>> bulkRename(Map<Path,Path> oldToNewPathMap, int poolSize,
+      String poolName, String transactionId) throws InterruptedException {
+    List<Future<Boolean>> results = new ArrayList<>();
+    SimpleThreadPool workerPool = new SimpleThreadPool(poolSize, poolName);
+    oldToNewPathMap.forEach((oldPath, newPath) -> results.add(workerPool.submit(() -> {
+      boolean success;
+      try {
+        success = rename(oldPath, newPath);
+      } catch (IOException e) {
+        // The rename could have failed because this is the second time its running (failures
+        // could cause this to run multiple times).
+        if (!exists(newPath) || exists(oldPath)) {
```

Comment:
I wonder if this check should also be done in the case where rename returns false.

Reply:
In that case would we throw a new IOException?

Reply:
I decided since we are already logging an error per file when rename returns false, we should not also throw a new error, failing the transation.  It seems like this is just to rethrow any errors we may care about.